spec/game_spec.rb

```diff
@@ -1,17 +1,118 @@
+    end
+
+    it 'when the variable `hint_clone_scode` is exist', positive: true do
+      expect(hint_clone_scode.class).to eq(Array)
+    end
+
+    it 'when the digit of `secret_code` is correct', positive: true do
+      expect(secret_code.size).to eq(DIGIT)
+    end
+
+    it 'when the digit of `secret_code` equal `hint_clone_scode` at the start', positive: true do
+      expect(secret_code.size == hint_clone_scode.size).to eq(true)
+    end
+
+    it 'when the sequence of `secret_code` is not same as `hint_clone_scode`  at the start', positive: true do
+      expect(game.hint_clone_scode).not_to eq(game.secret_code)
+    end
   end
-  context "comparing user input " do
-    it "check class of user input"
-    it "when user inpute equales secret code"
+
+  context 'when user did a coorect input of number' do
+    it 'when secret code equles user input, return game status `win`' do
+      user_input = secret_code.join
+      game.compare(user_input)
+      expect(game.compare(user_input)).to eq(MESSAGE_GU[:win])
+    end
+
+    it 'when user input differen with  secret code, return game status' do
+      expect(game.compare(NUMBER)).to be_instance_of(String)
+    end
   end
-  context " plus-minus factoring for user_input" do
-    it "check class of user input"
-    it "when answer_plus - array is created"
-    it "when answer_minus -hash is created"
+
+  context 'when the method plus-minus factoring output correct value' do
+    [
+      [[6, 5, 4, 1], [6, 5, 4, 1], '++++'],
+      [[1, 5, 3, 2], [5, 1, 3, 2], '++--'],
+      [[1, 2, 3, 4], [1, 3, 2, 4], '++--'],
+      [[1, 2, 3, 4], [1, 2, 4, 3], '++--'],
+      [[1, 2, 3, 4], [1, 4, 2, 3], '+---'],
+      [[1, 2, 3, 4], [4, 2, 1, 3], '+---'],
+      [[1, 2, 3, 4], [2, 4, 3, 1], '+---'],
+      [[1, 2, 3, 4], [2, 3, 1, 4], '+---'],
+      [[1, 2, 3, 4], [4, 3, 2, 1], '----'],
+      [[5, 4, 3, 2], [2, 3, 4, 5], '----'],
+      [[1, 2, 3, 4], [2, 1, 4, 3], '----'],
+      [[1, 3, 4, 2], [1, 2, 3, 4], '+---'],
+      [[5, 2, 5, 5], [2, 5, 5, 5], '++--'],
+      [[5, 5, 2, 5], [2, 5, 5, 5], '++--'],
+      [[5, 5, 5, 2], [2, 5, 5, 5], '++--'],
+      [[6, 2, 6, 2], [2, 6, 2, 6], '----'],
+      [[6, 6, 2, 2], [2, 6, 2, 6], '++--'],
+      [[2, 2, 6, 6], [2, 6, 2, 6], '++--'],
+      [[2, 6, 6, 2], [2, 6, 2, 6], '++--'],
+      [[6, 2, 2, 6], [2, 6, 2, 6], '++--'],
+      [[3, 1, 3, 5], [3, 3, 1, 5], '++--'],
+      [[3, 5, 1, 3], [3, 3, 1, 5], '++--'],
+      [[3, 3, 5, 1], [3, 3, 1, 5], '++--'],
+      [[1, 3, 5, 3], [3, 3, 1, 5], '+---'],
+      [[5, 3, 1, 3], [3, 3, 1, 5], '++--'],
+      [[1, 5, 3, 3], [3, 3, 1, 5], '----'],
+      [[5, 3, 3, 1], [3, 3, 1, 5], '+---'],
+      [[1, 2, 3, 4], [4, 3, 4, 3], '--']
+    ].each do |item|
+      it "when secret_code is #{item[0]} && the user input is #{item[1]}, the responds to consol will be #{item[2]}" do
+        allow(game).to receive(:compare).with(item[1].join)
+        game.instance_variable_set(:@secret_code, item[0])
```

Comment:
After this action  result of comparing is empty string

Reply:
You get an empty string because @answer_plus and @@answer_minus variables contain no values. You can add binding.pry here: https://github.com/MarynaYevnytska/codebreaker/pull/1/files#diff-1ffc50367eb4f4d631a2d2128996785cR67 and check it.
Also, there is no need for `allow` and `puts` in this test. It should look like:
```
it "when secret_code is #{item[0]} && the user input is #{item[1]}, the responds to consol will be #{item[2]}" do
        game.instance_variable_set(:@secret_code, item[0])
        expect(game.compare(item[1].join)).to eq(item[2])
      end
```